In [2]:
from glob import glob
import os

basePath = 'highway/valid/label/'

valid_filename = []
for filepath in glob(os.path.join(basePath, '*.xml')):
    filename = os.path.basename(filepath)  ## 상위폴더 경로 제거
    filename = filename[:-4]  ## .xml 제거
    valid_filename.append(filename)

len(valid_filename)

51

In [3]:
for folder in valid_filename:
    folderPath = os.path.join(basePath, folder)
    filepathList = glob(os.path.join(folderPath, '*'))
    print(len(filepathList))

150
150
50
50
150
230
150
100
50
150
230
150
100
100
150
149
150
150
625
150
150
150
150
50
150
150
150
100
50
50
150
150
150
625
150
150
150
150
625
147
149
150
150
625
150
150
150
150
624
31
224


In [4]:
basePath_image = 'highway/valid/image/'

for folder in valid_filename:
    folderPath = os.path.join(basePath_image, folder)
    filepathList = glob(os.path.join(folderPath, '*'))
    print(len(filepathList))

150
0
0
0
0
0
150
0
0
0
0
150
0
0
0
0
150
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0


### valid image 채널별 가로세로 사이즈 동일한지 확인

In [5]:
import cv2
import numpy as np


def glob_files(path, file_type="*"):
    search_string = os.path.join(path, file_type)
    files = glob(search_string)

    # print('searching ', path)
    paths = []
    for f in files:
      if os.path.isdir(f):
        sub_paths = glob_files(f + '/')
        paths += sub_paths
      else:
        paths.append(f)

    # We sort the images in alphabetical order to match them
    #  to the annotation files
    paths.sort()

    return paths


# IMAGE_SIZE = 600

def load_images(path):
  files = glob_files(path, "*.png")

  # print(files)
  X_data = []
  for file in files:
    image = cv2.imread(file)
    # print(image.shape)
    # x = cv2.resize(image, None, fx=0.5, fy=0.5, interpolation=cv2.INTER_AREA)

    X_data.append(image)
    
  return np.array(X_data)


for folder in valid_filename:
    folderPath = os.path.join(basePath_image, folder)
    X_test = load_images(folderPath)
    print(X_test.shape)

(150, 1920, 1080, 3)
(0,)
(0,)
(0,)
(0,)
(0,)
(150, 1080, 1920, 3)
(0,)
(0,)
(0,)
(0,)
(150, 1920, 1080, 3)
(0,)
(0,)
(0,)
(0,)
(150, 1920, 1080, 3)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)
(0,)


- CH02만 가로 1920 세로 1080 가로길쭉 -> transpose만 해주면 되나...?
- 나머지는 세로길쭉


In [6]:
X_test_ch02 = load_images(os.path.join(basePath_image, 'Suwon_CH02_20200720_2130_MON_9m_NH_highway_TW5_sunny_FHD'))

In [10]:
type(X_test_ch02)

numpy.ndarray

In [12]:
X_test_ch02.transpose().shape

(3, 1920, 1080, 150)

완전 뒤집어서 transpose 되어 버림.

In [14]:
X_test_ch02[0].transpose().shape

(3, 1920, 1080)

In [15]:
X_test_ch02.transpose((0, 2, 1, 3)).shape

(150, 1920, 1080, 3)

이렇게 하면 일단 이미지 사이즈 형식은 통일됨. -> ch02의 모든 박스도 transpose 해줘야 하나...?